rfd/0064-audit-modes.md

```diff
@@ -0,0 +1,190 @@
+---
+authors: Gabriel Corado (gabriel.oliveira@goteleport.com)
+state: draft
+---
+
+# RFD 64 - Audit modes
+
+## What
+
+This RFD proposes giving users a configuration to decide how to deal with
+sessions that can't be recorded or have audit logging failures.
+
+## Why
+
+In cases where Teleport is configured to record sessions on the node using the
+asynchronous format, disk failures (for example, no space left) causes the node
+to be inaccessible. Where Teleport is the only way of accessing those resources,
+administrators cannot perform the necessary actions on these nodes to bring them
+back to a normal state.
+
+## Goals
+
+* Prevent nodes from being inaccessible when there are audit errors.
+* Enable users to configure the strictness of audit failures.
+
+**Non-goals:**
+
+* Infrastructure-related metrics (such as disk space left).
+* Enable access when the recording mode is set to the proxy.
+
+## Details
+
+Audit modes will define how Teleport proceeds in case of audit failures. Those
```

Comment:
I would be specific with "audit" vs "session recording" terminology.

For audit events I think we only do "best effort" now since IIRC we basically ignore all errors from EmitAuditEvent (unless I'm mistaken). I would include it in the RFD also but focus on the session recording first to fix the original problem.

Reply:
Not in all cases. For example, in application access, [failing to emit events causes the request to fail](https://github.com/gravitational/teleport/blob/master/lib/srv/app/transport.go#L157).

Do you think we should also cover these cases?